Source/WebCore/page/make-user-agent-string-overrides.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2024 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+
+platforms = ['mac', 'iphone', 'ipad']
+user_agents = {
+    'chrome': '110.0.0.0',
+    'safari12': '12.1.2',
+    'safari13': '13.1.2',
+    'safari': '18.1',
+}
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate UserAgentOverride.h')
+    parser.add_argument('--input')
+    args = parser.parse_args()
+
+    # Version 1 input JSON file contains an object with:
+    #   - a version, currently the only supported value is the integer 1.
+    #   - objects where the key is a domain:
```

Comment:
We're going to need a way to specify if a quirk should apply to (a) the domain, exactly as specified, or (b) the toplevel base domain. E.g. one quirk for google.com might apply only to google.com and not also to drive.google.com, but another might need to apply to all subdomains of google.com.